src/trig.c

```diff
@@ -97,3 +97,102 @@ void twin_sincos(twin_angle_t a, twin_fixed_t *sin, twin_fixed_t *cos)
         *cos = cos_val;
     }
 }
+
+static const twin_angle_t atan_table[] = {
+    0x0200, /* arctan(2^0)  = 45° -> 512     */
+    0x0130, /* arctan(2^-1) = 26.565° -> 303 */
+    0x009B, /* arctan(2^-2) = 14.036° -> 155 */
+    0x004F, /* arctan(2^-3) = 7.125° -> 79   */
+    0x0027, /* arctan(2^-4) = 3.576° -> 39   */
+    0x0014, /* arctan(2^-5) = 1.790° -> 20   */
+    0x000A, /* arctan(2^-6) = 0.895° -> 10   */
+    0x0005, /* arctan(2^-7) = 0.448° -> 5    */
+    0x0002, /* arctan(2^-8) = 0.224° -> 2    */
+    0x0001, /* arctan(2^-9) = 0.112° -> 1    */
+    0x0001, /* arctan(2^-10) = 0.056° -> 1   */
+    0x0000, /* arctan(2^-11) = 0.028° -> 0   */
+};
+
+static twin_angle_t twin_atan2_first_quadrant(twin_fixed_t y, twin_fixed_t x)
+{
+    if (x == 0 && y == 0)
+        return 0;
+    if (x == 0)
+        return TWIN_ANGLE_90;
+    if (y == 0)
+        return TWIN_ANGLE_0;
+    twin_fixed_t current_x = x;
+    twin_fixed_t current_y = y;
+    twin_angle_t angle = 0;
+    /* CORDIC iteration */
+    for (int i = 0; i < 12; i++) {
```

Comment:
The iterative process only requires one temporary variable, temp_x, to calculate the angle, as following:

```c
twin_fixed_t temp_x;
for (int i = 0; i < 12; i++) {
    temp_x = current_x ;
    if (current_y > 0) {
        current_x += (current_y >> i);
        current_y -= (temp_x >> i);
        angle += atan_table[i];
    } else {
        current_x -= (current_y >> i);
        current_y += (temp_x >> i);
        angle -= atan_table[i];
    }
}
```